### PR TITLE
JS+JG US2-5 Complete

### PR DESCRIPTION
**Pull request by jrsewell400**
Made the following changes:
- Added tests to the navigation_spec file to check that links are correctly displayed to user based on their role.
- Added logic to application.html to handle displaying correct links based on user role.
- All added tests pass, but test coverage below 100%, see below:

**TEST COVERAGE UNDER 100%!**
Screen Shot 2020-04-08 at 4 53 47 PM

### CODE REVIEW

File: spec/features/navigation_spec.rb [hunk 1]
```diff
@@ -108,4 +127,118 @@
...
 end
+# As an admin
+# I see the same links as a regular user
+# Plus the following links
+
+# a link to my admin dashboard ("/admin")
+# a link to see all users ("/admin/users")
+# Minus the following links/info
+
+# a link to my shopping cart ("/cart") or count of cart items
```
Comment (elguapogordo) [inline spec/features/navigation_spec.rb:244]: i suspect these lines of code may be unnecessary

**Comment by elguapogordo** [comment]
i'm showing full coverage when i run the full set of tests

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -10,10 +10,26 @@
...
+        <% if current_user.merchant? %>
+          <%= link_to "Dashboard", "/merchant/dashboard" %>
+        <% elsif current_user.admin? %>
+          <%= link_to "Admin Dashboard", "/admin/dashboard" %>
+          <%= link_to "See all Users", "/admin/users" %>
+        <% end %>
+      <% else %>
+        <%= link_to "Register", "/register"%>
+        <%= link_to "Login", "/login"%>
+      <% end %>
```
Comment (zachholcomb) [inline app/views/layouts/application.html.erb:32]: view logic looks good, changed the user model to enums, these helper methods should still be available with enums

File: spec/features/navigation_spec.rb (hunk 1 above)
Comment (zachholcomb) [reply spec/features/navigation_spec.rb:244]: haha forgot to remove those

File: spec/features/navigation_spec.rb [hunk 3]
```diff
@@ -108,4 +127,118 @@
...
+        password: "123password",
+        role: 2
+      )
+
+      visit "/login"
+      fill_in :Email, with: admin.email
+      fill_in :Password, with: admin.password
+      click_button "Login"
+
+      within 'nav' do
```
Comment (zachholcomb) [inline spec/features/navigation_spec.rb:220]: do we need to click the ones that work or should we do that on later tests

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -10,10 +10,26 @@
...
-      <%= link_to "All Merchants", "/merchants"%>
+      <%= link_to "Home", "/"%>
       <%= link_to "All Items", "/items"%>
+      <%= link_to "All Merchants", "/merchants"%>
+      <% if !current_admin? %>
       <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Logout", "/logout", method: :delete %>
+      <% end %>
+      <% if current_user %>
+        <span>Logged in as <%= current_user.name %></span>
```
Comment (zachholcomb) [inline app/views/layouts/application.html.erb:20]: isn't their "username", the user's email? or should we go with name, think both are valid so don't have a preference really

**Comment by zachholcomb** [comment]
Looks good guys
Don't have too many comments, it's weird that Jordan's test coverage is so low, I want to pull this down and see if it's that way with mine too

File: spec/features/navigation_spec.rb (hunk 3 above)
Comment (jrsewell400) [reply spec/features/navigation_spec.rb:220]: Hmm, I think we could go either way with this.  The links we added - profile, merch dash, admin dash, and see all users - have either already been tested elsewhere (profile, merch dash, admin dash redirects in US15) and see all users will be built out in US53.  But it also kind of makes sense to make sure the nav bar functionality of these links work, too.

I can add the link tests for nav in my US21 work, not a very heavy lift to add that.

File: spec/features/navigation_spec.rb (hunk 1 above)
Comment (jrsewell400) [reply spec/features/navigation_spec.rb:244]: OOPS - I'll remove that in my US21 branch!

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (jrsewell400) [reply app/views/layouts/application.html.erb:20]: Based on the user story, it looks like they wanted the person's name:

_"I also see text that says "Logged in as Mike Dao" (or whatever my name is)"_

**Pull request merged**